src/algebra/field.lean

```diff
@@ -214,46 +214,73 @@ end
 
 end
 
-namespace is_ring_hom
-open is_ring_hom
+namespace ring_hom
 
 section
-variables {β : Type*} [division_ring α] [division_ring β]
-variables (f : α → β) [is_ring_hom f] {x y : α}
+
+variables {β : Type*} [division_ring α] [division_ring β] (f : α →+* β) {x y : α}
 
 lemma map_ne_zero : f x ≠ 0 ↔ x ≠ 0 :=
-⟨mt $ λ h, h.symm ▸ map_zero f,
- λ x0 h, one_ne_zero $ calc
-    1 = f (x * x⁻¹) : by rw [mul_inv_cancel x0, map_one f]
-  ... = 0 : by rw [map_mul f, h, zero_mul]⟩
+⟨mt $ λ h, h.symm ▸ f.map_zero,
+ λ x0 h, one_ne_zero $ by rw [← f.map_one, ← mul_inv_cancel x0, f.map_mul, h, zero_mul]⟩
 
 lemma map_eq_zero : f x = 0 ↔ x = 0 :=
-by haveI := classical.dec; exact not_iff_not.1 (map_ne_zero f)
+by haveI := classical.dec; exact not_iff_not.1 f.map_ne_zero
 
 lemma map_inv' (h : x ≠ 0) : f x⁻¹ = (f x)⁻¹ :=
-(domain.mul_left_inj ((map_ne_zero f).2 h)).1 $
-by rw [mul_inv_cancel ((map_ne_zero f).2 h), ← map_mul f, mul_inv_cancel h, map_one f]
+(domain.mul_left_inj (f.map_ne_zero.2 h)).1 $
+by rw [mul_inv_cancel (f.map_ne_zero.2 h), ← f.map_mul, mul_inv_cancel h, f.map_one]
 
 lemma map_div' (h : y ≠ 0) : f (x / y) = f x / f y :=
-(map_mul f).trans $ congr_arg _ $ map_inv' f h
+(f.map_mul _ _).trans $ congr_arg _ $ f.map_inv' h
```

Comment:
@digama0 Is it time to make changes to Lean core, and fix the definition of field and division ring?

Reply:
Fine with me, although I would prefer to delegate the scheduling of the first non-backward compatible lean release to @robertylewis and @cipher1024 .